src/storage/src/store_impl.rs

```diff
@@ -566,75 +570,116 @@ impl StateStoreImpl {
         compactor_metrics: Arc<CompactorMetrics>,
         await_tree_config: Option<await_tree::Config>,
     ) -> StorageResult<Self> {
-        set_foyer_metrics_registry(GLOBAL_METRICS_REGISTRY.clone());
+        const MB: usize = 1 << 20;
+
+        set_metrics_registry(GLOBAL_METRICS_REGISTRY.clone());
+
+        let meta_cache_v2 = {
+            let mut builder = HybridCacheBuilder::new()
+                .memory(opts.meta_cache_capacity_mb * MB)
+                .with_shards(opts.meta_cache_shard_num)
+                .with_eviction_config(opts.meta_cache_eviction_config.clone())
+                .with_object_pool_capacity(1024 * opts.meta_cache_shard_num)
+                .with_weighter(|_: &HummockSstableObjectId, value: &Box<Sstable>| {
+                    u64::BITS as usize / 8 + value.estimate_size()
+                })
+                .storage();
+
+            if !opts.meta_file_cache_dir.is_empty() {
+                builder = builder
+                    .with_name("foyer.meta")
+                    .with_device_config(
+                        FsDeviceConfigBuilder::new(&opts.meta_file_cache_dir)
+                            .with_capacity(opts.meta_file_cache_capacity_mb * MB)
+                            .with_file_size(opts.meta_file_cache_file_capacity_mb * MB)
+                            .with_align(opts.meta_file_cache_device_align)
+                            .with_io_size(opts.meta_file_cache_device_io_size)
+                            .build(),
+                    )
+                    .with_catalog_shards(64)
```

Comment:
Is it expected hardcode to '64' ?

Reply:
IMO, 64 is an optimized value for most RW cases. We can make it configurable after we face a problem.